secretsauce/apps/account/views.py

```diff
@@ -10,62 +11,43 @@
 from djoser.conf import settings
 from djoser.views import UserViewSet
 
-from secretsauce.apps.account.models import Invitation, Company
-from secretsauce.apps.account.serializers import InvitationSerializer, CompanySerializer
+from secretsauce.apps.account.models import *
+from secretsauce.apps.account.serializers import *
+from secretsauce.utils import *
 
-class InvitationCreator(mixins.CreateModelMixin,
-                        mixins.ListModelMixin,
-                        generics.GenericAPIView):
-    """
-    Create a new invitation
-    """
+class CompanyList(generics.ListCreateAPIView):
 
     permission_classes = [permissions.IsAdminUser]
-    queryset = Invitation.objects.all()
-    serializer_class = InvitationSerializer
+    queryset = Company.objects.all()
+    serializer_class = CompanySerializer
 
-    def create(self, request, *args, **kwargs):
-        serializer = self.serializer_class(data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            email = serializer.data.get('email')
-            invitation_id = serializer.data.get('id')
-            message = "Your invitation code is " + invitation_id + "."
-            send_mail('Invitation to the RMS Club', message, 'sutdcapstone22@gmail.com', [email], fail_silently=False)
-            return Response(serializer.data, status=status.HTTP_201_CREATED)
-        elif 'invitation with this email already exists.' in serializer.errors['email']:
-            email = serializer.data.get('email')
-            invitation_id = str(Invitation.objects.get(email=email).id)
-            message = "Your invitation code is " + invitation_id + "."
-            send_mail('Invitation to the RMS Club', message, 'sutdcapstone22@gmail.com', [email], fail_silently=False)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+class CompanyDetail(generics.RetrieveUpdateDestroyAPIView):
 
-    def post(self, request, *args, **kwargs):
-        return self.create(request, *args, **kwargs)
+    permission_classes = [permissions.IsAdminUser]
+    queryset = Company.objects.all()
+    serializer_class = CompanySerializer
 
-    def get(self, request, *args, **kwargs):
-        return self.list(request, *args, **kwargs)
+class InviteUser(generics.CreateAPIView):
 
-class InvitationDetail(mixins.RetrieveModelMixin,
-                       generics.GenericAPIView):
-    """
-    Check invitation validity
-    """
+    permission_classes = [permissions.IsAdminUser]
 
-    queryset = Invitation.objects.all()
-    serializer_class = InvitationSerializer
+    def create(self, request):
+        password = generatePassword()
+        user_email = request.data['email']
+        user = User.objects.create_user(
+            user_email,
+            password
+        )
+        user.save()
+        serializer=UserCreateSerializer(user)
```

Comment:
The creation of users is abit insecure, because there are no checks being done on the data being sent. For example, I can call "email=qweqwe", which is not a valid email, but an account is created. 

Also, I am able to send in the email of an existing account, which overwrites the existing account.

Reply:
Hmmm I thought django's EmailField would help to validate that the email provided is a valid email and coz it's supposed to be unique, it wont allow for overriding. Let me look into this.

Reply:
I think call User.objects.create(...) instead of create_user(...)

Reply:
Actually I'm not too sure about that ^. But maybe can try adding the serializer_class to do the field checks

Reply:
https://stackoverflow.com/questions/6039443/primary-key-and-unique-key-in-django
Can create a PR for this feature on django page LOL